File: domain/src/main/java/com/example/app/booking/domain/Seat.java
package com.example.app.booking.domain;

import com.example.app.common.exception.APIException;
import com.example.app.movie.type.TheaterSeat;
import lombok.Builder;

import java.time.LocalDate;
import java.util.List;

import static com.example.app.booking.exception.BookingErrorMessage.SEAT_ALREADY_OCCUPIED;

@Builder
public record Seat(
        Long id,
        Long bookingId,
        Long movieId,
        Long showtimeId,
        Long theaterId,
        LocalDate bookingDate,
        TheaterSeat theaterSeat,
        boolean reserved
) {

    public static void checkSeatsAvailable(List<Seat> seats) {
        for (Seat seat : seats) {
            if (seat.reserved()) {
                throw new APIException(SEAT_ALREADY_OCCUPIED);
            }
        }
    }
}

File: movie/core/src/main/kotlin/com/example/movie/domain/seat/model/Seat.kt
package com.example.movie.domain.seat.model

import com.example.movie.domain.common.model.BaseModel
import com.example.movie.domain.theater.model.Theater
import java.time.LocalDateTime

class Seat(
    val id: Long,
    val theater: Theater,
    val row: Char,
    val col: Int,
    override val createdBy: String,
    override val createdAt: LocalDateTime,
    override var updatedBy: String,
    override var updatedAt: LocalDateTime
) : BaseModel